src/components/filters/getFilterProps.js

```diff
@@ -28,38 +30,58 @@ export default function getFilterProps(column, descriptions, filters, setFilter)
   throw new Error(`unreachable: ${description.type}`)
 }
 
-function getInputFilterProps(column, descriptions, filters, setFilter) {
+function getInputFilterProps(column, descriptions, filters, setFilter, setFilterOption) {
   const dataIndex = column.dataIndex;
   const description = descriptions[dataIndex];
+  const value = filters[dataIndex]?.value;
 
   const inputRef = useRef()
 
-  const onSearch = (selectedKeys, setSelectedKeys) => {
-    setSelectedKeys(selectedKeys)
-    setFilter(dataIndex, selectedKeys[0])
+  const onSearch = value => {
+    setFilter(dataIndex, value)
   }
 
   const onKeyDown = (ev, confirm) => {
     if (ev.key === 'Escape')
       confirm()
   }
 
+  const onToggleSwitch = checked => {
+    setFilterOption(dataIndex, 'exactMatch', checked)
+  }
+
+  const selectedValue = filters => {
+    return filters[dataIndex]?.value?.[0]
+  }
+
+  const selectedExactMatch = filters => {
+    return filters[dataIndex]?.options?.exactMatch
+  }
```

Comment:
These 2 above don't need to be functions, they select a single value once, they can be variables. Also at the start of `getInputFilterProps` we're already assigning the value to `const value`, we can probably reuse that one and extract `options` at the same place.